app/features/send/spark-lightning-send-service.ts

```diff
@@ -0,0 +1,250 @@
+  constructor(private readonly sparkWallet: SparkWallet) {}
+
+  /**
+   * Estimates the fee for paying a Lightning invoice and returns a quote for the send.
+   */
+  async getLightningSendQuote({
+    paymentRequest,
+    amount,
+    getExchangeRate,
+  }: GetSparkLightningSendQuoteOptions): Promise<SparkLightningSendQuote> {
+    const bolt11ValidationResult = parseBolt11Invoice(paymentRequest);
+    if (!bolt11ValidationResult.valid) {
+      throw new DomainError('Invalid lightning invoice');
+    }
+    const invoice = bolt11ValidationResult.decoded;
+
+    if (invoice.expiryUnixMs && new Date(invoice.expiryUnixMs) < new Date()) {
+      throw new DomainError('Lightning invoice has expired');
+    }
+
+    let amountRequestedInBtc = new Money({
+      amount: 0,
+      currency: 'BTC',
+    });
+
+    if (invoice.amountMsat) {
+      amountRequestedInBtc = new Money({
+        amount: invoice.amountMsat,
+        currency: 'BTC',
+        unit: 'msat',
+      });
+    } else if (amount) {
+      if (amount.currency === 'BTC') {
+        amountRequestedInBtc = amount as Money<'BTC'>;
+      } else if (getExchangeRate) {
+        // TODO: see if we need this. Our send-store already picks the currency based on the account.
+        // We do a similar conversion for cashu send quotes so I did it here, but SPark accounts can only be BTC anyway.
+        const exchangeRate = await getExchangeRate('USD-BTC');
+        amountRequestedInBtc = amount.convert('BTC', exchangeRate);
+      } else {
+        throw new Error('Exchange rate is required for non-BTC amounts');
+      }
+    } else {
+      throw new Error('Unknown send amount');
+    }
+
+    const estimatedLightningFeeSats =
+      await this.sparkWallet.getLightningSendFeeEstimate({
+        amountSats: amountRequestedInBtc.toNumber('sat'),
+        encodedInvoice: paymentRequest,
+      });
+
+    const estimatedLightningFee = new Money({
+      amount: estimatedLightningFeeSats,
+      currency: 'BTC',
+      unit: 'sat',
+    });
+
+    const estimatedTotalAmount = amountRequestedInBtc.add(
+      estimatedLightningFee,
+    );
+
+    return {
+      paymentRequest,
+      amountRequested: amount ?? (amountRequestedInBtc as Money),
+      amountRequestedInBtc,
+      amountToReceive: amountRequestedInBtc as Money,
+      estimatedLightningFee,
+      estimatedTotalFee: estimatedLightningFee as Money,
+      estimatedTotalAmount: estimatedTotalAmount as Money,
+      paymentRequestIsAmountless: invoice.amountMsat === undefined,
+    };
+  }
+
+  /**
+   * Pays a Lightning invoice.
+   * @throws Error if the payment fails or if the fee exceeds maxFeeSats
+   */
+  async initiateSend({
+    quote,
+  }: CreateSparkLightningSendParams): Promise<SparkLightningSend> {
+    const request = await this.sparkWallet.payLightningInvoice({
+      invoice: quote.paymentRequest,
+      maxFeeSats: quote.estimatedLightningFee.toNumber('sat'),
+      preferSpark: false,
```

Comment:
does this make it use Spark internal transfer instead of lightning? if yes why would we not want to do it? it would be cheaper and faster right?